src/app/vm/vm-creation/components/affinity-group-manager/vm-creation-affinity-group-manager.component.ts

```diff
@@ -0,0 +1,68 @@
+import { Component, EventEmitter, forwardRef, Input, Output } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { MatDialog, MatDialogConfig } from '@angular/material';
+import { AffinityGroup } from '../../../../shared/models';
+import { AffinityGroupSelectorContainerComponent } from '../../../vm-sidebar/affinity-group-selector/affinity-group-selector-container.component';
+
+@Component({
+  selector: 'cs-vm-creation-affinity-group-manager',
+  templateUrl: 'vm-creation-affinity-group-manager.component.html',
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => VmCreationAffinityGroupManagerComponent),
+      multi: true,
+    },
+  ],
+  styleUrls: ['vm-creation-affinity-group-manager.component.scss'],
+})
+export class VmCreationAffinityGroupManagerComponent implements ControlValueAccessor {
```

Comment:
Why implement `ngModel` if you don't use it? You only use affinityGroupChanged EventEmitter.